src/movies.js

```diff
@@ -1,25 +1,85 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+
+function getAllDirectors(moviesArray) {
+    const justDirectors = moviesArray.map((movies) => movies.director);
+    return justDirectors;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    const dramaMovies = moviesArray.filter((movies) => movies.director === "Steven Spielberg" && movies.genre.includes("Drama"));
+        return dramaMovies.length;
+    }
+    
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+
+    const scores = moviesArray.reduce((acc, val) => {
+        if(val.score){
+            return acc + val.score;     
+        } else{
+            return acc;
+        }
+  }, 0);  
+
+   return Math.round ((scores / moviesArray.length) * 100) / 100;
+   
+}
+   
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
```

Comment:
This iteration is not passing this test `"should only calculate the average for drama movies"`
In the previous iteration you already created a function that calculates the average (scoresAverage). So you can use it inside this new function, and in this way the only thing left to do is check if the movie includes the genre drama:

```
function dramaMoviesScore(moviesArray) {
  const dramaScores = moviesArray.filter((movie) =>
    movie.genre.includes("Drama")
  );
  return scoresAverage(dramaScores);
}
```